test/com/linkedin/parseq/TestTasks.java

```diff
@@ -89,6 +91,53 @@ public void onResolved(Promise<String> stringPromise) {
     assertEquals(Boolean.TRUE, resultRef.get());
   }
 
+  @Test
+  public void testToTask() {
+    final AtomicReference<Boolean> resultRef = new AtomicReference<Boolean>(false);
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        resultRef.set(true);
+        return "";
+      });
+      return completableFuture;
+    });
+    runAndWait("testToTask.success", task);
+    assertEquals(Boolean.TRUE, resultRef.get());
+  }
+
+  @Test
+  public void testToTaskWithFailure() {
+    final AtomicReference<Boolean> resultRef = new AtomicReference<Boolean>(false);
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        try {
+          Thread.sleep(10000);
+        } catch (InterruptedException e) {
+          throw new RuntimeException(e);
+        }
+        resultRef.set(true);
+        return "";
+      });
+      return completableFuture;
+    });
+    run(task);
+    assertEquals(Boolean.FALSE, resultRef.get());
+  }
+
+  @Test
+  public void testToTaskWithException() {
+    Task<String> task = Task.fromFuture(() ->{
+      CompletableFuture<String> completableFuture
+          = CompletableFuture.supplyAsync(() -> {
+        throw new RuntimeException();
+      });
+      return completableFuture;
+    });
+    runAndWaitException(task, CompletionException.class);
+  }
+
```

Comment:
We also need a test where Callable throws an Exception.

Reply:
Add in the to test